stdlib/sql/sql_test.flux

```diff
@@ -13,42 +14,58 @@ pgDsn = "postgresql://postgres@127.0.0.1:5432/postgres?sslmode=disable"
 verticaDsn = "vertica://dbadmin@localhost:5433/flux"
 sqliteDsn = "file:///tmp/flux-integ-tests-sqlite.db"
 
-stanley = {name: "Stanley", age: 15}
-lucy = {name: "Lucy", age: 14}
-sophie = {name: "Sophie", age: 15}
+// Some db engines will UPPERCASE table/column names when the identifiers are unquoted.
+// At time of writing the DDL used to create the tables has unquoted column names, so
+// for engines such as Snowflake and SAP HANA (hdb) we need to UPPERCASE these identifiers
+// here so the quoted version matches the table definition.
+stanley = {name: "Stanley", age: 15, "fav food": "chicken"}
+STANLEY = {NAME: "Stanley", AGE: 15, "FAV FOOD": "chicken"}
 
-// Some db engines will UPPERCASE table/column names when the identifiers are unquoted
-STANLEY = {NAME: "Stanley", AGE: 15}
+lucy = {name: "Lucy", age: 14}
 LUCY = {NAME: "Lucy", AGE: 14}
-SOPHIE = {NAME: "Sophie", AGE: 15}
+
+sophie = {name: "Sophie", age: 15, "fav food": "salmon"}
+SOPHIE = {NAME: "Sophie", AGE: 15, "FAV FOOD": "salmon"}
+
+nonseed_want = array.from(rows: [sophie])
+NONSEED_WANT = array.from(rows: [SOPHIE])
 
 testcase integration_hdb_read_from_seed {
-    want = array.from(rows: [STANLEY, LUCY])
-    got = sql.from(driverName: "hdb", dataSourceName: hdbDsn, query: "SELECT name, age FROM pets WHERE seeded = true")
+    got =
+        sql.from(
+            driverName: "hdb",
+            dataSourceName: hdbDsn,
+            // n.b. we must explicitly UPPER CASE the table name here to match the DDL.
+            query: "SELECT name, age, \"FAV FOOD\" FROM \"PET INFO\" WHERE seeded = true",
+        )
 
-    testing.diff(got: got, want: want)
+    testing.diff(
+        got: got,
+        want:
+            union(tables: [array.from(rows: [STANLEY]) |> debug.opaque(), array.from(rows: [LUCY]) |> debug.opaque()])
```

Comment:
I had originally wanted to have all the "want" unions to be defined at the head of the flux script, but the test harness started reporting "0 of 0 tests" when I did this.

I tried wrapping them in functions but got the same result.

Reply:
That seems strange, and doesn't fit with my understanding of how tests are run. I would be curious to know the test case that gets that behavior.

Reply:
I can try and work up a minimal repro case. I'd sort of gotten used to bumping into these edges and just dismissing as me doing something odd, but if that's not the case I can open a separate issue for it.

Reply:
So, whatever was happening before is no longer happening today. I suspect there was a syntax error in the flux script someplace and I was mistaken about the cause. I'm pushing a rev that hoists these unions up to the top of the script. :clap: